src/groupeddataframe/splitapplycombine.jl

```diff
@@ -194,24 +194,27 @@ function _combine_process_agg(@nospecialize(cs_i::Pair{Int, <:Pair{<:Function, S
                               gd::GroupedDataFrame,
                               seen_cols::Dict{Symbol, Tuple{Bool, Int}},
                               trans_res::Vector{TransformationResult},
-                              idx_agg::Union{Nothing, AbstractVector{Int}})
+                              idx_agg::Ref{Union{Nothing, Vector{Int}}})
     @assert isagg(cs_i, gd)
     @assert !optional_i
     out_col_name = last(last(cs_i))
     incol = parentdf[!, first(cs_i)]
     agg = check_aggregate(first(last(cs_i)), incol)
     outcol = agg(incol, gd)
 
-    if haskey(seen_cols, out_col_name)
-        optional, loc = seen_cols[out_col_name]
-        # we have seen this col but it is not allowed to replace it
-        optional || throw(ArgumentError("duplicate output column name: :$out_col_name"))
-        @assert trans_res[loc].optional && trans_res[loc].name == out_col_name
-        trans_res[loc] = TransformationResult(idx_agg, outcol, out_col_name, optional_i)
-        seen_cols[out_col_name] = (optional_i, loc)
-    else
-        push!(trans_res, TransformationResult(idx_agg, outcol, out_col_name, optional_i))
-        seen_cols[out_col_name] = (optional_i, length(trans_res))
+    return function()
+        if haskey(seen_cols, out_col_name)
```

Comment:
A small question to check (sorry for being picky) - have you checked if we do not have problems with boxing in this anonymous function (same below). It is just curiosity, as even if we do boxing it is not a problem as we do not do any expensive operations inside.

Reply:
I've pushed an alternative version which returns a function an arguments rather than an anonymous function to nl/threadedops2. It's actually slightly slower. It could be considered as a bit cleaner but it's more verbose. Let me know what you think.

Reply:
I prefer this one 😄. It actually seems easier to read and follow the logic. Even better that it is faster.